### ISSUE TITLE
Intermittent problem with IQN setting

### ISSUE THREAD

**Issue by frederictobiasc** [OpenMediaVault-Plugin-Developers/openmediavault-tgt#9]
Hi
when creating a new target, sometimes the plugin fails to set the IQN correctly ([see video](https://user-images.githubusercontent.com/26125115/147606000-d83bcee9-2d88-4d41-aa1b-cbda709b5721.mp4)).
This results in an error when trying to apply the configuration changes. Tried on a fresh installation from today on armbian bullseye.

tgt-plugin version: 6.0.2
OMV- version: 6.0.6-1 (Shaitan)
Processor: ARMv7 Processor rev 3 (v7l)
Kernel: Linux 5.4.160-odroidxu4

**Comment by ryecoaaron** [comment]
Could you put omv-engined in debug mode and see if you can see errors?
As root:
monit stop omv-engined
omv-engined -df

**Comment by frederictobiasc** [comment]
Sure.
Similar to the video, `iqn` is set when first created. When modified (set enable), `iqn` is set to "".
```
omv-engined[4133]: Executing RPC (service=Tgt, method=setTarget, params={"uuid":"fa4b1c66-ef79-11e5-87a0-0002b3a176b4","enable":false,"name":"another_test","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":""}, context={"username":"admin","role":1}) ...
omv-engined[4133]: 
omv-engined[4133]: RPC response (service=Tgt, method=setTarget): {"response":{"uuid":"7faa69cc-9459-49a0-a3da-7bf177efb882","enable":false,"name":"another_test","iqn":"iqn.2021-12.nas-schoeneweiss:another_test","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":""},"error":null}
omv-engined[4133]: 
omv-engined[4058]: SIGCHLD received ...
omv-engined[4058]: 
omv-engined[4058]: Child (pid=4133) terminated with exit code 0
omv-engined[4058]: 
omv-engined[4142]: Child process forked (pid=4142)
omv-engined[4142]: 
omv-engined[4142]: Executing RPC (service=Tgt, method=getTargetList, params={"start":0,"limit":-1}, context={"username":"admin","role":1}) ...
omv-engined[4142]: 
omv-engined[4142]: RPC response (service=Tgt, method=getTargetList): {"response":{"total":3,"data":[{"uuid":"31722ad8-bee6-47a3-a50a-83baf527685b","enable":true,"name":"testf","iqn":"iqn.2021-12.odroidxu4:testf","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/testf","initiatoraddress":"","extraoptions":"","initiators":""},{"uuid":"acb873fb-60ba-4c04-a6fc-3f4ae65f456f","enable":true,"name":"test","iqn":"iqn.2021-12.odroidxu4:test","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":"","initiators":""},{"uuid":"7faa69cc-9459-49a0-a3da-7bf177efb882","enable":false,"name":"another_test","iqn":"iqn.2021-12.nas-schoeneweiss:another_test","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":"","initiators":""}]},"error":null}
omv-engined[4142]: 
omv-engined[4058]: SIGCHLD received ...
omv-engined[4058]: 
omv-engined[4058]: Child (pid=4142) terminated with exit code 0
omv-engined[4058]: 
omv-engined[4143]: Child process forked (pid=4143)
omv-engined[4143]: 
omv-engined[4143]: Executing RPC (service=System, method=getInformation, params=null, context={"username":"admin","role":1}) ...
omv-engined[4143]: 
omv-engined[4143]: RPC response (service=System, method=getInformation): {"response":{"ts":1640768943,"time":"Wed Dec 29 10:09:03 2021","hostname":"nas-schoeneweiss","version":"6.0.6-1 (Shaitan)","cpuModelName":"ARMv7 Processor rev 3 (v7l)","cpuUsage":0,"memTotal":"2087145472","memFree":"46612480","memUsed":"170008576","memAvailable":"1797058560","memUtilization":"0.13898","kernel":"Linux 5.4.160-odroidxu4","uptime":5021.18,"loadAverage":{"1min":0.81,"5min":0.69,"15min":0.64},"configDirty":true,"rebootRequired":false,"pkgUpdatesAvailable":false},"error":null}
omv-engined[4143]: 
omv-engined[4058]: SIGCHLD received ...
omv-engined[4058]: 
omv-engined[4058]: Child (pid=4143) terminated with exit code 0
omv-engined[4058]: 
omv-engined[4152]: Child process forked (pid=4152)
omv-engined[4152]: 
omv-engined[4152]: Executing RPC (service=Tgt, method=getTarget, params={"uuid":"7faa69cc-9459-49a0-a3da-7bf177efb882"}, context={"username":"admin","role":1}) ...
omv-engined[4152]: 
omv-engined[4152]: RPC response (service=Tgt, method=getTarget): {"response":{"uuid":"7faa69cc-9459-49a0-a3da-7bf177efb882","enable":false,"name":"another_test","iqn":"iqn.2021-12.nas-schoeneweiss:another_test","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":""},"error":null}
omv-engined[4152]: 
omv-engined[4058]: SIGCHLD received ...
omv-engined[4058]: 
omv-engined[4058]: Child (pid=4152) terminated with exit code 0
omv-engined[4058]: 
omv-engined[4213]: Child process forked (pid=4213)
omv-engined[4213]: 
omv-engined[4213]: Executing RPC (service=Tgt, method=setTarget, params={"uuid":"7faa69cc-9459-49a0-a3da-7bf177efb882","enable":true,"name":"another_test","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":""}, context={"username":"admin","role":1}) ...
omv-engined[4213]: 
omv-engined[4213]: RPC response (service=Tgt, method=setTarget): {"response":{"uuid":"7faa69cc-9459-49a0-a3da-7bf177efb882","enable":true,"name":"another_test","iqn":"","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":""},"error":null}
omv-engined[4213]: 
omv-engined[4058]: SIGCHLD received ...
omv-engined[4058]: 
omv-engined[4058]: Child (pid=4213) terminated with exit code 0
omv-engined[4058]: 
omv-engined[4214]: Child process forked (pid=4214)
omv-engined[4214]: 
omv-engined[4214]: Executing RPC (service=Tgt, method=getTargetList, params={"start":0,"limit":-1}, context={"username":"admin","role":1}) ...
omv-engined[4214]: 
omv-engined[4214]: RPC response (service=Tgt, method=getTargetList): {"response":{"total":3,"data":[{"uuid":"31722ad8-bee6-47a3-a50a-83baf527685b","enable":true,"name":"testf","iqn":"iqn.2021-12.odroidxu4:testf","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/testf","initiatoraddress":"","extraoptions":"","initiators":""},{"uuid":"acb873fb-60ba-4c04-a6fc-3f4ae65f456f","enable":true,"name":"test","iqn":"iqn.2021-12.odroidxu4:test","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":"","initiators":""},{"uuid":"7faa69cc-9459-49a0-a3da-7bf177efb882","enable":true,"name":"another_test","iqn":"","backingstore":"\/srv\/dev-disk-by-uuid-23e54a5b-e08b-49a9-af88-cffc88d9e84e\/backup-nb-regina\/test","initiatoraddress":"","extraoptions":"","initiators":""}]},"error":null}
```

**Comment by ryecoaaron** [comment]
This should be fixed.  I will push out a new package when I fix issue #10
